### PR TITLE
T016 automatic qgis project creation

### PR DESCRIPTION
**Pull request by Fred1511**

### CODE REVIEW

File: src/core/session.py [hunk 1]
```diff
@@ -47,7 +49,17 @@ def onCreateSession(self, directory):
...
-        QgsProject.instance().addMapLayer(self._gpsTable)
+        uri = (
+            "geopackage:"
+            + SammoDataBase.pathToDataBase(directory)
+            + "?projectName=sammo_boat_project"
+        )
+        if isNewDataBase:
+            # Save the QGIS projet into the database
+            QgsProject.instance().addMapLayer(self._gpsTable)
+            QgsProject.instance().write(uri)
```
Comment (pblottiere) [inline src/core/session.py:60]: We should create a new project from scratch in this case instead of using the current one.

Comment (Fred1511) [reply src/core/session.py:61]: Searching for the solution, I found the clear() method (from QgsProject class) which "_Clear the project - removes all settings and resets it back to an empty, default state._". I think it is the quickest way to achieve what you want.

Comment (pblottiere) [reply src/core/session.py:61]: You can just create a whole new project with `project = QgsProject()` instead of using the current instance

**Comment by pblottiere** [comment]
Hello @Fred1511,

Can you rebase please? Because I have a strange error with this branch when I start the GPS simulator:

``` bash
src/core/qgsmessagelog.cpp:29 : (logMessage) [7987ms] 2021-09-02T14:43:46 Python error[1] Traceback (most recent call last):
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/default/python/plugins/sammo-boat/src/core/session.py", line 138, in addNewFeatureToGpsTable
    self._gpsTable.startEditing()
RuntimeError: wrapped C/C++ object of type QgsVectorLayer has been deleted
```

**Comment by Fred1511** [comment]
We have to keep in mind that, after merging t016 and t020 into main, we will have to configure transect layer so it will be automatically updated on map as well (cf. last commit).

**Comment by pblottiere** [comment]
Hello @Fred1511,

Can you rebase please? Because there's some conflicts which need to be fixed now after merging the last PR.

**Comment by pblottiere** [comment]
Hi @Fred1511. I just tried to create a new session with this branch but I have the next error:

``` python
AttributeError: 'SammoSimuGpsBtn' object has no attribute 'onNewSession' 
Traceback (most recent call last):
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/sammo.py", line 153, in projectLoaded
    self._simuGpsBtn.onNewSession()
AttributeError: 'SammoSimuGpsBtn' object has no attribute 'onNewSession'
```

Any idea?

**Comment by pblottiere** [comment]
I have another error now ^^:

``` python
AttributeError: 'SammoSession' object has no attribute 'onCreateSession' 
Traceback (most recent call last):
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/sammo.py", line 103, in onCreateSession
    self._session.onCreateSession(workingDirectory)
AttributeError: 'SammoSession' object has no attribute 'onCreateSession'
```

**Comment by pblottiere** [comment]
Thanks @Fred1511 :+1:

**Pull request merged**